### PR TITLE
Remove ethgetblock task

### PR DESCRIPTION
**Pull request by martin-cll**
Remove the `ethgetblock` task. This task was only used by the Mercury OCR2 plugin which replaces this task with an inline call to get the latest block.

The task run results are extracted by the `datasource` into a mercury relay observation in the `Observe` method. I added a new `setCurrentBlock` method here to perform the same logic as the removed task, i.e. fetch latest block from the HeadTracker, otherwise falling back to making an eth call. This requires a `evm.Chain` which is passed down from the ocr2 delegate.

Because the final values are no longer exclusively derivable from the task run results using the `parse` method, I've had to shuffle some variables around the `executeRun`, `parse` and `collectMercuryEnhancedTelemetry` methods. Now, `executeRun` returns all task run results (not just final ones) and `parse` accepts all task run results (again, not just the final ones). I've also pulled the call to `collectMercuryEnhancedTelemetry` into the outer `Observe` method so that it can accept the Observation with the new block number and hash.

### CODE REVIEW

File: core/services/relay/evm/mercury/data_source.go [hunk 1]
```diff
@@ -48,7 +46,15 @@ func (ds *datasource) Observe(ctx context.Context, repts ocrtypes.ReportTimestam
...
 	}
 
-	return ds.parse(trrs)
+	obs, err := ds.parse(trrs)
+	if err != nil {
+		return relaymercury.Observation{}, fmt.Errorf("Observe failed while parsing run results: %w", err)
+	}
+	ds.setCurrentBlock(ctx, &obs)
+
+	go collectMercuryEnhancedTelemetry(ds, &trrs, obs, repts)
```
Comment (samsondav) [inline core/services/relay/evm/mercury/data_source.go:55]: this seems unrelated?

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/services/ocr2/plugins/mercury/plugin.go [hunk 2]
```diff
@@ -52,6 +53,7 @@ func NewServices(
 		lggr,
 		runResults,
 		me,
+		chain,
```
[the remark this answers is not in the document]
Comment (martin-cll) [reply core/services/ocr2/plugins/mercury/plugin.go:56]: Replaced it with a new ChainHeadTracker ifc.

File: core/services/relay/evm/mercury/data_source.go (hunk 1 above)
Comment (martin-cll) [reply core/services/relay/evm/mercury/data_source.go:55]: This line used to be called at the end of `ds.executeRun` which is at the beginning of this function. I pulled it out of executeRun and put it here instead. The reason being that I wanted to pass in the observation into collectMercuryEnhancedTelemetry rather than have collectMercuryEnhancedTelemetry fetch the latest block again (previously it calls `ds.parse` a second time to get values from the task run results which is fine as it doesn't involve IO, but now it could involve an eth call which I want to avoid).

**Review by samsondav — DISMISSED** [review]

File: core/services/ocr2/delegate.go [hunk 3]
```diff
@@ -351,7 +359,7 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 			OnchainKeyring:         kb,
 		}
 		me := d.monitoringEndpointGen.GenMonitoringEndpoint(spec.ContractID, synchronization.EnhancedEAMercury)
-		return mercury.NewServices(jb, mercuryProvider, d.pipelineRunner, runResults, lggr, oracleArgsNoPlugin, d.cfg, me)
+		return mercury.NewServices(jb, mercuryProvider, d.pipelineRunner, runResults, lggr, oracleArgsNoPlugin, d.cfg, me, chain)
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:362]: Using an EVM specific `ChainHeadTracker` type reintroduces the original problem we are trying to avoid here. We need to encapsulate this logic inside of the `mercuryProvider`, and only interact directly with the chain-agnostic API/types from here.

Comment (ainsleybc) [reply core/services/ocr2/delegate.go:362]: Tech debt will be address in https://smartcontract-it.atlassian.net/browse/MERC-619

**Review by jmank88 — APPROVED** [review]
